assets/src/components/AssignmentTable.js

```diff
@@ -120,170 +144,272 @@ function AssignmentTable (props) {
     }
   }, [currentWeekRow.current])
 
+  const [assignmentGroupNames, setAssignmentGroupNames] = useState([])
+  useEffect(() => {
+    const allGroupNames = assignments.map(ag => ag.assignmentGroup.name)
+    var uniqueGroupNames = [...new Set(allGroupNames)]
+    uniqueGroupNames.sort()
+    setAssignmentGroupNames(uniqueGroupNames)
+  }, [assignments])
+
+  const [assignmentFilter, setAssignmentFilter] = useState('')
+  function handleNameFilterChange (e) {
+    setAssignmentFilter(e.target.value)
+  }
+
+  const ITEM_HEIGHT = 48
+  const ITEM_PADDING_TOP = 8
+  const MenuProps = {
+    PaperProps: {
+      style: {
+        maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+        width: 250
+      }
+    }
+  }
+
+  const [assignmentGroupFilterArray, setAssignmentGroupFilterArray] = useState([])
+  function handleGroupNameArrayFilterChange (e) {
+    setAssignmentGroupFilterArray(e.target.value)
+  }
+
+  function handleGradedOnlyFilterChange (e) {
+    setGradedOnly(!gradedOnly)
+  }
```

Comment:
Nit Pick: The functions like above in various places can be rewritten as below. this is consistence with the pattern used in the code

```
const handleGradedOnlyFilterChange = e => {
setGradeOnly(!gradedOnly)
}
```

Reply:
Fixed